### PR TITLE
Update `createGithubIssue` to re-open recently closed issue

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
Update `createGithubIssue` to re-open recently closed issue

### JenkinsFile
```
lib = library(identifier: 'jenkins@main', retriever: modernSCM([
    $class: 'GitSCMSource',
    remote: 'https://github.com/prudhvigodithi/opensearch-build-libraries.git',
]))

pipeline {
    options {
        timeout(time: 4, unit: 'HOURS')
    }
    agent none
    stages {
        stage('gh-test') {
            agent {label 'Jenkins-Agent-AL2023-X64-C54xlarge-Docker-Host'}
            steps {
                script {
                    createGithubIssue(
                        repoUrl: "https://github.com/prudhvigodithi/job-scheduler.git",
                        issueTitle: "[RELEASE] Release version 3.10.0",
                        issueBody: "test library",
                        label: "v3.9.0",
                        daysToReOpen: "10" // Optional: Tested enabling and disabling 
                    )
                }
            }
        }
    }
} 
```

### New issue

https://github.com/prudhvigodithi/job-scheduler/issues/27

### Comment on existing issue

https://github.com/prudhvigodithi/job-scheduler/issues/26#issuecomment-1785872564

### Re-open and comment on recently closed issue 

Re-open: https://github.com/prudhvigodithi/job-scheduler/issues/26#event-10812010651
Comment: https://github.com/prudhvigodithi/job-scheduler/issues/26#issuecomment-1785873226

### Issues Resolved

https://github.com/opensearch-project/opensearch-build/issues/4171
Coming from the logic https://github.com/opensearch-project/opensearch-build/issues/4171#issuecomment-1778397737 

#### When passed `daysToReOpen: "10"` goes back to 10 days from current date `2023-10-20`

Screenshot 2023-10-30 at 4 10 34 PM

#### When now passed `daysToReOpen` uses default 3 days from current date `2023-10-27`

Screenshot 2023-10-30 at 4 11 07 PM

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: vars/createGithubIssue.groovy [hunk 1]
```diff
@@ -14,26 +14,46 @@
...
+
+            if (openIssue) {
+                println('Issue already exists, adding a comment')
+                sh(
+                   script: "gh issue comment ${openIssue} --repo ${args.repoUrl} --body \"${args.issueBody}\"",
+                   returnStdout: true
+                )
+            }
+            else if (!openIssue && closedIssue) {
+                println("Commenting on recently closed issue")
```
Comment (gaiksaya) [inline vars/createGithubIssue.groovy:45]:
```suggestion
                println("Re-opening a recently closed issue and commenting on it")
```

File: vars/createGithubIssue.groovy [hunk 2]
```diff
@@ -14,26 +14,46 @@
...
-                println('Issue already exists, adding a comment.')
-                def issuesNumber = sh(
+            def openIssue = sh(
                     script: "gh issue list --repo ${args.repoUrl} -S \"${args.issueTitle} in:title\" --label ${label} --json number --jq '.[0].number'",
                     returnStdout: true
-                ).trim()
+            ).trim()
+
+            def currentDayMinusThree = sh(
+                script: "date -d \"3 days ago\" +'%Y-%m-%d'",
```
Comment (gaiksaya) [inline vars/createGithubIssue.groovy:28]: Do you want to take this as an argument? How many days ago do we want to re-open issues for? Maybe just take the integer as an arg with default value as 3?

Comment (prudhvigodithi) [reply vars/createGithubIssue.groovy:28]: I thought so,  but again if we have to give that option to user, we should go with complete regex meaning
`daysToReOpen: 3 days ago` or `daysToReOpen: 6 hours ago` or `daysToReOpen: closed in last 2 months` or even more. This should be compatible with linux `date` and with github search query (converting back to yyyy-MM-dd format). Just giving an option to use to change in days might not be very useful.

Comment (gaiksaya) [reply vars/createGithubIssue.groovy:28]: We can just start with integer value fixed to number of days. WDYT? I am assuming we will soon need to change this too so avoiding another release just to fix that 1 value

File: vars/createGithubIssue.groovy (hunk 1 above)
Comment (prudhvigodithi) [reply vars/createGithubIssue.groovy]: Updated in my latest commit.

File: vars/createGithubIssue.groovy (hunk 2 above)
Comment (prudhvigodithi) [reply vars/createGithubIssue.groovy:28]: Ya I can have it hardcoded in days, but the number of days user can configure through the library. Will make that change soon.

Comment (prudhvigodithi) [reply vars/createGithubIssue.groovy]: @gaiksaya I have updated the code in my latest commit, also update the description with testing information.

File: vars/createGithubIssue.groovy [hunk 3]
```diff
@@ -13,27 +13,49 @@
...
-            if (issues) {
-                println('Issue already exists, adding a comment.')
-                def issuesNumber = sh(
+            def openIssue = sh(
                     script: "gh issue list --repo ${args.repoUrl} -S \"${args.issueTitle} in:title\" --label ${label} --json number --jq '.[0].number'",
                     returnStdout: true
-                ).trim()
+            ).trim()
+
+            def currentDayMinusThree = sh(
```
Comment (gaiksaya) [inline vars/createGithubIssue.groovy:29]: Maybe rename this variable. Sounds confusing. 
```suggestion
            def currentDayMinusDaysToReOpen = sh(
```

Comment (prudhvigodithi) [reply vars/createGithubIssue.groovy:29]: Looks like this is a NIT, the variable `currentDayMinusThree` is a `Date` and then used in a search query. 
The name `currentDayMinusThree` makes more sense as the output of this variable is `Date` but this `currentDayMinusDaysToReOpen` does not look like `Date`.

@gaiksaya  how about `dateThreeDaysAgo` ?

Comment (gaiksaya) [reply vars/createGithubIssue.groovy:29]: My point is `Three` might not be true right? Since we are taking arg from the user too.

Comment (prudhvigodithi) [reply vars/createGithubIssue.groovy]: Updated @gaiksaya

**Review by gaiksaya — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**